messaginginapp/src/main/java/io/customer/messaginginapp/gist/presentation/GistModalActivity.kt

```diff
@@ -32,17 +33,22 @@ const val GIST_MODAL_POSITION_INTENT: String = "GIST_MODAL_POSITION"
 @InternalCustomerIOApi
 class GistModalActivity : AppCompatActivity(), ModalInAppMessageViewCallback, TrackableScreen {
     private lateinit var binding: ActivityGistBinding
-    private var elapsedTimer: ElapsedTimer = ElapsedTimer()
-    private val inAppMessagingManager = SDKComponent.inAppMessagingManager
-    private val state: InAppMessagingState
-        get() = inAppMessagingManager.getCurrentState()
-    private val logger = SDKComponent.logger
-    private val attributesListenerJob: MutableList<Job> = mutableListOf()
-
     private var messagePosition: MessagePosition = MessagePosition.CENTER
 
+    private val attributesListenerJob: MutableList<Job> = mutableListOf()
+    private val elapsedTimer: ElapsedTimer = ElapsedTimer()
+    private val logger = SDKComponent.logger
+
+    // Dependencies requiring ModuleMessagingInApp to be initialized must be accessed lazily,
+    // only after confirming the SDK has been initialized.
+    private val inAppMessagingManager: InAppMessagingManager?
+        get() = kotlin.runCatching {
+            SDKComponent.inAppMessagingManager
+        }.getOrNull()
```

Comment:
I wonder can we consider initializing it in `onCreate`? lazy works but brings the risk on data races. 

```
    private var inAppMessagingManager: InAppMessagingManager? = null
    
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        }
 ```

Reply:
`inAppMessagingManager` is already a thread safe singleton, so I believe this shouldn't be an issue?